src/server/game/Battlegrounds/ArenaTeam.cpp

```diff
@@ -317,6 +318,25 @@ void ArenaTeam::DelMember(ObjectGuid guid, bool cleanDb)
     // Remove member from team
     for (MemberList::iterator itr = Members.begin(); itr != Members.end(); ++itr)
     {
+        // Remove queues of members
+        if (Player* player = ObjectAccessor::FindPlayer(itr->Guid))
```

Comment:
I wonder what happens if one player is being teleported to the arena while the other leaves the party

Reply:
Deadlock or nullptr exception, maybe.

Reply:
@jackpoz, this seems better to you: https://github.com/TrinityCore/TrinityCore/pull/21491/commits/425b8a82c495e0cf49ad2b4c8b214eeaae00e11f?